ignore/ignore.go

```diff
@@ -0,0 +1,106 @@
+package ignore
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+)
+
+const IGNORE_FILE_NAME = ".farnsworthignore"
+
+// A Filter determines whether or not a given path should be
+// ignored.
+type Filter interface {
+	ShouldIgnore(filePath string) bool
+}
+
+// A predicate returns `true` for paths that should be excluded from
+// an archive or merge operation, and `false` otherwise.
+//
+// For example, users might want to keep version control metadata out of
+// archives, in which case a path that ended in `.gitignore` might cause
+// the predicate to return `true`.
+type predicate func(path string) bool
+
+// A Filter is a collection of Predicates.
+type filter struct {
+	predicates []predicate
+}
+
+// ShouldIgnore returns the logical disjunction of the predicates
+// included in the Filter.
+func (f *filter) ShouldIgnore(filePath string) bool {
+	for _, predicate := range f.predicates {
+		if predicate(filePath) {
+			return true
+		}
+	}
+	return false
+}
+
+// addPredicate adds a predicate to the Filter.
+func (f *filter) addPredicate(predicate predicate) {
+	f.predicates = append(f.predicates, predicate)
+}
+
+// newRegexPredicate creates a predicate that checks to see if the path
+// matches the given regular expression. The predicate will return
+// `true` if they match.
+func newRegexPredicate(patternString string) (predicate, error) {
+	pattern, err := regexp.CompilePOSIX(patternString)
```

Comment:
I'm curious if compiling in the function is ideal. Maybe the function did take the pattern interface.